### PR TITLE
lib: nrf_modem: update doc and samples with Zephyr UART trace backend

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Use Zephyr UART trace backend in doc and samples as the other UART trace backends are depreceated.

### CODE REVIEW

**Review by lemrey — APPROVED** [review]

File: doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst [hunk 1]
```diff
@@ -8,11 +8,11 @@ Modem trace module
...
-The module is implemented in :file:`nrf/lib/nrf_modem_lib/nrf_modem_lib_trace.c` and consists of a thread that initializes, deinitializes, and forwards modem traces to a backend that can be selected by enabling any one of the following Kconfig options:
+The module is implemented in :file:`nrf/lib/nrf_modem_lib/nrf_modem_lib_trace.c` and consists of a thread that initializes, deinitializes, and forwards modem traces to a backend that can be selected by enabling any one of the following Kconfig options and snippets:
 
-* :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_UART` to send modem traces over UART
-* :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RTT` to send modem traces over SEGGER RTT
-* :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_FLASH` to write modem traces to external flash
+* ``nrf91-modem-trace-uart`` snippet to send modem traces over UART
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RTT` Kconfig option to send modem traces over SEGGER RTT
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_FLASH` Kconfi option to write modem traces to external flash
```
Comment (simensrostad) [inline doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst:15]:
```suggestion
* :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_FLASH` Kconfig option to write modem traces to external flash
```

**Review by simensrostad — APPROVED** [review]
Does this change require updates to the cellular monitor apps startup message?

image

**Comment by eivindj-nordic** [comment]
> Does this change require updates to the cellular monitor apps startup message?
> 
> image

Yes. I've sent a heads-up to Ådne (don't know the github user) some time ago.

File: doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst (hunk 1 above)
Comment (eivindj-nordic) [reply doc/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.rst:15]: thanks, updated.

**Review by anhmolt — APPROVED** [review]

File: samples/net/aws_iot/README.rst [hunk 2]
```diff
@@ -225,7 +225,7 @@ Troubleshooting
 To enable more verbose logging from the AWS IoT library, enable the :kconfig:option:`CONFIG_AWS_IOT_LOG_LEVEL_DBG` option.
 
 * If you are having issues with connectivity on nRF91 Series devices, see the `Trace Collector`_ documentation to learn how to capture modem traces in order to debug network traffic in Wireshark.
-  The sample enables modem traces by default, as set by the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE` option.
+  The sample enables modem traces by default.
```
Comment (divipillai) [inline samples/net/aws_iot/README.rst:228]: Shouldn't we mention about the ``nrf91-modem-trace-uart`` snippet in this case?

File: samples/net/mqtt/doc/description.rst [hunk 3]
```diff
@@ -275,7 +275,7 @@ Troubleshooting
 To enable more verbose logging from the MQTT helper library, enable the :kconfig:option:`CONFIG_MQTT_HELPER_LOG_LEVEL_DBG` option.
 
 * If you are having issues with connectivity on nRF91 Series devices, see the `Trace Collector`_ documentation to learn how to capture modem traces in order to debug network traffic in Wireshark.
-  The sample enables modem traces by default, as set by the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE` option.
+  The sample enables modem traces by default.
```
Comment (divipillai) [inline samples/net/mqtt/doc/description.rst:278]: Shouldn't we mention about the nrf91-modem-trace-uart snippet in this case?

File: samples/net/aws_iot/README.rst (hunk 2 above)
Comment (eivindj-nordic) [reply samples/net/aws_iot/README.rst:228]: We're not adding the snippet in this case, but adding the config to the sample itself. I've updated the release notes to reflect this.

File: samples/net/mqtt/doc/description.rst (hunk 3 above)
Comment (eivindj-nordic) [reply samples/net/mqtt/doc/description.rst:278]: We're not adding the snippet in this case, but adding the config to the sample itself. I've updated the release notes to reflect this.

**Review by divipillai — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

File: doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst [hunk 4]
```diff
@@ -156,17 +156,16 @@ Modem trace
 
 The modem traces of the nRF9160 modem can be captured using the nRF Connect Trace Collector.
 For more information on how to collect traces using nRF Connect Trace Collector, see the `Trace Collector`_ documentation.
-When the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE` Kconfig option is enabled, the modem traces are enabled in the modem and are forwarded to the :ref:`modem_trace_module`.
+To enable the modem traces in the modem and to forward to the :ref:`modem_trace_module` over UART, include the ``nrf91-modem-trace-uart`` snippet while building your application as described in :ref:`nrf91_modem_trace_uart_snippet`.
```
Comment (MirkoCovizzi) [inline doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst:159]:
```suggestion
To enable the modem traces in the modem and to forward them to the :ref:`modem_trace_module` over UART, include the ``nrf91-modem-trace-uart`` snippet while building your application as described in :ref:`nrf91_modem_trace_uart_snippet`.
```

**Review by MirkoCovizzi — APPROVED** [review]

**Pull request merged**